saleor/tests/e2e/orders/test_order_in_channel_with_transaction_flow_as_mark_as_paid_strategy.py

```diff
@@ -0,0 +1,155 @@
+    permission_manage_channels,
+    permission_manage_product_types_and_attributes,
+    permission_manage_shipping,
+    permission_manage_orders,
+    permission_manage_payments,
+):
+    # Before
+    permissions = [
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_orders,
+        permission_manage_payments,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+    app_permissions = [permission_manage_payments]
+    assign_permissions(e2e_app_api_client, app_permissions)
+
+    (
+        warehouse_id,
+        channel_id,
+        shipping_method_id,
+    ) = prepare_shop_in_channel_with_transaction_flow_as_mark_as_paid_strategy(
+        e2e_staff_api_client
+    )
+
+    price = 2
+    (
+        _,
+        result_product_variant_id,
+        _,
+    ) = prepare_product(e2e_staff_api_client, warehouse_id, channel_id, price)
+
+    # Step 1 - Create draft order
+    draft_order_input = {
+        "channelId": channel_id,
+        "userEmail": "test_user@test.com",
+        "shippingAddress": DEFAULT_ADDRESS,
+        "billingAddress": DEFAULT_ADDRESS,
+    }
+    data = draft_order_create(
+        e2e_staff_api_client,
+        draft_order_input,
+    )
+    order_id = data["order"]["id"]
+    assert order_id is not None
+
+    # Step 2 - Add lines to the order
+    lines = [{"variantId": result_product_variant_id, "quantity": 1}]
+    order_lines = order_lines_create(e2e_staff_api_client, order_id, lines)
+    order_product_variant_id = order_lines["order"]["lines"][0]["variant"]["id"]
+    assert order_product_variant_id == result_product_variant_id
+
+    # Step 3 - Update order's shipping method
+    input = {"shippingMethod": shipping_method_id}
+    draft_order = draft_order_update(e2e_staff_api_client, order_id, input)
+    order_shipping_id = draft_order["order"]["deliveryMethod"]["id"]
+    assert order_shipping_id is not None
+
+    # Step 4 - Complete the order and check it's status
+    order = draft_order_complete(e2e_staff_api_client, order_id)
+    order_complete_id = order["order"]["id"]
+    assert order_complete_id == order_id
+    order_line = order["order"]["lines"][0]
+    assert order_line["productVariantId"] == result_product_variant_id
+    assert order["order"]["status"] == "UNFULFILLED"
+    assert (
+        order["order"]["channel"]["orderSettings"]["markAsPaidStrategy"]
+        == "TRANSACTION_FLOW"
+    )
+    order_total = order["order"]["total"]["gross"]["amount"]
+    assert order_total != 0
+
+    # Step 5 - Create transaction and check order's payment status
+    create_transaction(
+        e2e_app_api_client,
+        order_id,
+        transaction_name="CreditCard",
+        message="Charged",
+        psp_reference="PSP-ref123",
+        available_actions=["REFUND", "CANCEL"],
+        amount=order_total,
+    )
```

Comment:
For making an order as paid in transaction flow you need to run a mutation orderMarkAsPaid with transaction reference and then check if the created order.transactions is not empty.
```
mutation OrderMarkAsPaid($id: ID!, $transactionReference: String) {
  orderMarkAsPaid(id: $id, transactionReference: $transactionReference) {
    errors {
      code
    }
    order {
      id
    }
  }
}



{
  "id": "id",
  "transactionReference": "test-pspreference"
}
```